### PR TITLE
chore: Issue warning announcing entity's value_type as mandatory 

### PR DESCRIPTION
**Pull request by franciscojavierarceo**
# What this PR does / why we need it:
Make value_type for Entity issue a warning if not set. Next release we'll make it mandatory.

# Which issue(s) this PR fixes:
https://github.com/feast-dev/feast/issues/4670

# Misc

### CODE REVIEW

File: sdk/python/feast/entity.py [hunk 1]
```diff
@@ -79,6 +80,13 @@ def __init__(
             ValueError: Parameters are specified incorrectly.
         """
         self.name = name
+        if value_type is None:
+            warnings.warn(
+                "Entity value_type will be mandatory in the next release. "
+                "Please specify a value_type for entity '%s'." % name,
+                DeprecationWarning,
```
Comment (dmartinol) [inline sdk/python/feast/entity.py:87]: why not FutureWarning instead?
https://docs.python.org/3/library/exceptions.html#FutureWarning

Comment (franciscojavierarceo) [reply sdk/python/feast/entity.py:87]: DeprecationWarning:
>Base class for warnings about deprecated features when those warnings are intended for other Python developers.
>
>Ignored by the default warning filters, except in the __main__ module ([PEP 565](https://peps.python.org/pep-0565/)). Enabling the [Python Development Mode](https://docs.python.org/3/library/devmode.html#devmode) shows this warning.

FutureWarning
>Base class for warnings about deprecated features when those warnings are intended for end users of applications that are written in Python.

I think DeprecationWarning is a reasonable choice but I could see both ways. Devin chose the former. 🤷

Comment (dmartinol) [reply sdk/python/feast/entity.py:87]: I can’t see any deprecation actually, but if Devin will become the main maintainer, that’s fine 🤷🏻‍♂️

Comment (franciscojavierarceo) [reply sdk/python/feast/entity.py:87]: I'm going to force it next release.

**Review by dmartinol — APPROVED** [review]

**Pull request merged**